### ISSUE TITLE
Duplicate patterns appear while training on custom dataset

### ISSUE THREAD

**Issue by AshAJ1994** [jwmao1/MedSegFactory#16]
Hi Jwmao1, 

Great work !
Thanks for sharing the code :) 

Hi, I've been trying to fine-tuning MedSegFactory on a single medical imaging dataset ~ 7000 image + mask (single class) pair (not the full multi-dataset setup from the paper,) and I'm seeing a consistent artifact where the generated images show duplicate/double structures — the target anatomy appears as two overlapping or stacked bands rather than one.

Training setup:

Mode: finetune (loading from runwayml/stable-diffusion-v1-5)
Learning rate: 1e-5
Batch size: 4
Resolution: 512
Mixed precision: fp16
Single dataset, single class segmentation (similar to Polyp dataset mentioned in your dataset)
Loss values observed:

Is this artifact known when training on a single dataset vs. the full multi-dataset mixture from the paper? 
Are there any known training tricks (e.g., specific LR schedule, dropout rates, dataset augmentation) that help with structural duplication artifacts in generated output? Please let me know if you would want to share more details to suggest potential workaround. 

Any guidance would be appreciated. Thanks again for the great work on this paper.

<img width="2244" height="326" alt="Image" src="https://github.com/user-attachments/assets/4d8eb0c8-e704-4903-b208-803515f7a672" />

**Comment by jwmao1** [comment]
Hi, sorry for the late reply, and thanks for your interest in our work. Could you share a few example results showing the duplicated structure artifact? That would help us better understand the problem. Thanks again for the feedback.

**Comment by AshAJ1994** [comment]
Hi @jwmao1 , 

Apologies for the delayed response. 

After a bit of digging in the code, found that original collate_fn uses aspect ratio selection with a hardcoded max_size=256. 

aspect = aspect_ratios[random.randint(0, len(aspect_ratios) - 1)]
shape = get_target_size(aspect, 256)

I gave input --resolution = 512. So probably the model tiles the 256-trained pattern to fill the 512 canvas — producing the dual band artifact. Please correct me if I am wrong here. 

**Follow-up questions:**

1. Conditioning on mask-derived quantitative measurements:
My dataset includes paired image-mask samples where each mask has a corresponding quantitative measurement (e.g. average  layer thickness or area in pixels). Is it feasible to encode this scalar directly into the text prompt — e.g. "..., average thickness 87 pixel" — so the model learns the joint relationship between image, mask, and the numeric descriptor? Or does CLIP's text encoder lack the numerical grounding for this to work reliably, in which case what alternative conditioning approach would you recommend?

2. Loss and hyperparameter guidance for fine-tuning: 
Training loss starts at 0.077 from epoch 1 : is this expected or a case of model overfitting ? 
What loss range and curve shape should be expected for healthy fine-tuning on a single-class dataset vs. the full multi-dataset mixture? And are there recommended hyperparameter optimization strategies ? 

Thanks a lot for this work ! Appreciate your support.

**Comment by jwmao1** [comment]
Hi @AshAJ1994 , thanks for the detailed check.

Yes, this is likely a resolution mismatch issue. We recommend using `--resolution=512`; `--resolution=512` may introduce preprocessing inconsistency. 

For scalar conditioning, adding `"average thickness 87 pixels"` to the prompt is okay as a baseline, but CLIP text conditioning may not learn precise numeric control reliably. For better control, I’d suggest either binning the value or adding a small MLP-based scalar conditioning branch.

For the loss, `0.077` from epoch 1 is not necessarily overfitting since the model starts from pretrained weights. I’d monitor sample quality, image-mask alignment, and diversity rather than relying only on the absolute loss. Single-class fine-tuning may overfit faster, so smaller LR, fewer epochs, augmentation, and checkpoint comparison are recommended.

**Comment by AshAJ1994** [comment]
Hi @jwmao1 , 

Thanks for the response and for your inputs on scalar conditioning , hyper parameter tuning.

**Comment by jwmao1** [comment]
Thanks again for your questions and for your interest in our work. I really appreciate your feedback!
